arch-design/delfin/RequirementsDesignSpec.md

```diff
@@ -0,0 +1,278 @@
+
+
+
+**Performance Monitoring**
+
+Register storage for performance monitoring
+Report key performance indicators periodically 
+
+**Alert Monitoring** 
+
+Real time reporting of events/alerts.
+
+Pull alerts from back end on user request.
+
+Clear Alert on user request.
+
+
+#### Requirement Analysis
+Analyzed various Open Source and Enterprise monitoring solutions.
+
+
+Conclusion:-
+There is no open source framework for addressing all requirements in one platform.
+
+
+
+#### List of Requirements
+##### Functional Requirements
+* Registering a storage device
+* Removing a storage device
+* Get registered storage device list
+* Updating storage device access information
+* Querying storage device details
+* Querying other resource  Information like (LUN,Pool,Disk, etc)
+* Provide performance metrics of different resources
+* Sync all registered  device collection
+* Sync particular  device  collection
+* Register alert source to receive alerts
+* Report alerts
+* Clear alert
+
+
+##### Non Functional Requirements
+**Performance requirements**
+
+NA
+**Security requirements**
+1. Support secured access to delfin APIS
+2. Support secured access to back-ends
+
+**Other non functional requirements (scalability  , HA etc…)**
+NA
+
+
+## Architecture Analysis
+### System Architecture
+![Architecture](./resources/Architecture.jpg)
+
+### Module Architecture
+NA
+
+### Architecture Tenets
+**North-bound API** :- REST interface to delfin
+
+**Exporter interface**:- Python class interfaces to push information out of delfin. 
+
+**Driver interface**:- Python class interfaces to be implemented by driver to collect information's from back end. 
+
+**Database** :- Any relational database.
+
+**API server or Resource Manager** :- Process which serves the REST APIs
+
+**Task manager**:- Process which prepares, schedule and update collection jobs.
+
+**Driver**:- An Object which is part of other processes to connect and collect information’s from a back end.
+
+**Exporter**:- An object which is part of task process to push data out.
+
+### High Level Module Architecture
+
+Task manager module architecture [here](https://github.com/sodafoundation/architecture-analysis/blob/43d1020021e12b6e0dc1833f05d4159623c0ac6e/SIM/RetrievalManagement.md)
+
+Alert Manager module architecture [here](https://github.com/sodafoundation/architecture-analysis/blob/8549aa6ca5bf7296dedbba7b62ef63e0550a83f2/specs/SIM/alert_manager/SODA_AlertManagerDesign.md) 
+
+Driver Manager module architecture [here](https://github.com/sodafoundation/architecture-analysis/blob/694d4aa31dd4577aca5a67668b96f615c4f34be9/specs/SIM/DriverManager/DriverManager.md) 
+
+
+## Detailed Design
+
+### Use case View
+
+![Usecase](resources/Delfin_usecase.jpg)
+
+#### List of Typical Use-cases
+
+1. **Adding Storage Devices for monitoring**:- 	User call APIS by specifying  device access information such as EMC VMax, Hitachi Virtual Storage Platform, or HP 3PAR etc... 
```

Comment:
alignment is inconsistent with rest of the point and ":-" can be removed

Reply:
Done